plugins/inputs/prometheus/prometheus.go

```diff
@@ -273,17 +284,27 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 		return fmt.Errorf("error reading body: %s", err)
 	}
 
-	metrics, err := Parse(body, resp.Header)
+	if p.MetricVersion == 2 {
+		metrics, err = ParseV2(body, resp.Header)
+	} else {
+		metrics, err = Parse(body, resp.Header)
+	}
+
 	if err != nil {
 		return fmt.Errorf("error reading metrics for %s: %s",
 			u.URL, err)
 	}
 
+	var urltag = "url"
```

Comment:
If you set the default value in the `init()` function to `url`, then an explicit empty string can mean not to add the tag.

Reply:
fixed this.